src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,194 @@
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    }).then(() => {
+      getAllContacts()
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      } else {
+        getAllContacts()
+      }
+    }
+  }, [])
+
+  const handleSearch = useCallback(query => {
+    const queryIsNumber = parseInt(query)
+    if (state) {
+      setValue(query)
+    }
+    if (query && !query.includes('+') && !query.includes('*')) {
+      if (typeof queryIsNumber === 'number' && !isNaN(queryIsNumber)) {
+        return setContacts(
+          initialList.filter(({ phoneNumbers }) => get(phoneNumbers, '[0].number', '').indexOf(query) >= 0)
+        )
+      }
+      if (typeof query === 'string') {
+        return setContacts(
+          initialList.filter(({ givenName, familyName }) => {
+            const fullName = `${givenName} ${familyName}`.toLocaleLowerCase()
+            return fullName.indexOf(query.toLocaleLowerCase()) >= 0
+          })
+        )
+      }
+    } else {
+      getAllContacts()
+    }
+  })
+
+  useEffect(() => {
+    if (contacts.length === 0) {
+      showNext(true)
+    } else {
+      showNext(false)
+    }
+  }, [contacts])
+
+  useEffect(() => {
+    findRecentlyUsed()
+  }, [contacts, recentFeedItems])
+
+  const findRecentlyUsed = () => {
+    let matches = []
+    if (recentFeedItems.length > 0 && contacts.length > 0) {
+      recentFeedItems.forEach(feedItem => {
+        matches.push(
+          contacts.filter(
+            contact => contact.phoneNumbers[0] && feedItem.data.phoneNumber === contact.phoneNumbers[0].number
+          )
+        )
+      })
+    }
+
+    setRecentlyUsedList(matches.flat())
+  }
+
+  return (
+    <>
+      <Section.Stack justifyContent="space-around" style={styles.container}>
+        <Section.Title fontWeight="medium">{text}</Section.Title>
+        <InputText
+          error={error}
+          onChangeText={handleSearch}
+          placeholder="Search contact name / phone"
+          style={styles.input}
+          value={value}
+          enablesReturnKeyAutomatically
+          iconName="search"
+        />
+      </Section.Stack>
+      {recentlyUsedList.length > 0 && (
+        <>
+          <Section.Row justifyContent="space-between">
+            <Section.Title fontWeight="medium" style={styles.sectionTitle}>
+              {'Recently used'}
+            </Section.Title>
+            <Section.Separator style={styles.separator} width={1} />
+          </Section.Row>
+          <Section.Row>
+            <FlatList
+              data={uniq(recentlyUsedList).slice(0, 5)}
+              renderItem={({ item, index }) => (
```

Comment:
- renderItem could be memoized via useCallback
- ItemSeparatorComponent also could be defined as separate component and moved outside WhoContent

```javascript
const Separator = withStyles(() => ({
  separator: {
     flex: 1,
     opacity: 0.3,
   },
})(({ styles }) => (
  <Separator color={styles.separatorColor} />
))

// ...
const WhoContent = ...

  const renderItem = useCallback(({ item, index }) => (
    <FeedContactItem contact={item} selectContact={setContact} horizontalMode index={index} />
  ), [setContact])

  // ...
  <FlatList
               
    renderItem={renderItem}
    ItemSeparatorComponent={Separator}
```